src/framework/components/rigid-body/system.js

```diff
@@ -522,38 +529,16 @@ Object.assign(pc, function () {
             return r && (r.hasEvent("collisionstart") || r.hasEvent("collisionend") || r.hasEvent("contact"));
         },
 
-        onUpdate: function (dt) {
-            // #ifdef PROFILER
-            this._stats.physicsStart = pc.now();
-            // #endif
-
-            // Check to see whether we need to update gravity on the dynamics world
-            var gravity = this.dynamicsWorld.getGravity();
-            if (gravity.x() !== this.gravity.x || gravity.y() !== this.gravity.y || gravity.z() !== this.gravity.z) {
-                gravity.setValue(this.gravity.x, this.gravity.y, this.gravity.z);
-                this.dynamicsWorld.setGravity(gravity);
-            }
-
-            // Update the transforms of all bodies
-            this.dynamicsWorld.stepSimulation(dt, this.maxSubSteps, this.fixedTimeStep);
-
-            // Update the transforms of all entities referencing a body
-            var components = this.store;
-            for (var id in components) {
-                if (components.hasOwnProperty(id)) {
-                    var entity = components[id].entity;
-                    var componentData = components[id].data;
-                    if (componentData.body && componentData.body.isActive() && componentData.enabled && entity.enabled) {
-                        if (componentData.type === pc.BODYTYPE_DYNAMIC) {
-                            entity.rigidbody.syncBodyToEntity();
-                        } else if (componentData.type === pc.BODYTYPE_KINEMATIC) {
-                            entity.rigidbody._updateKinematic(dt);
-                        }
-                    }
-
-                }
-            }
-
+        /**
+         * @private
+         * @function
+         * @name pc.RigidBodyComponentSystem#_checkForCollisions
+         * @description Checks for collisions and fires collision events
+         * @param {pc.Entity} dynamicsWorld - The pointer to the dynamics world that invoked this callback.
```

Comment:
Type is wrong. Should be `object`.

Reply:
Also, I would say 'The dynamics world that...` instead of `The pointer to the dynamics world...` since this is JavaScript rather than C++. 😄

Reply:
Given the point about the integer pointer, would @param {object | number} be appropriate here?

Reply:
Sounds like it should be `number` 😄

Reply:
If it's passed in when this function is called from the update function for versions of Ammo that don't have setInternalTickCallback then it will be an object.